### ISSUE TITLE
JSON-API serialisation failure due to resource hash with nested data key

### ISSUE THREAD

**Issue by john-griffin** [emberjs/data#3381]
See https://github.com/emberjs/data/commit/76cf421556c80cff61462c49cd70cc193bdca9c5#commitcomment-11749944

When fetching a JSON-API 1.0 compliant resource it fails because serializer expects data to be at the top level of the `resourceHash` when it is actually nested in `resourceHash.data`.

Might be fixable by ensuring `resourceHash.data` is passed through here? https://github.com/emberjs/data/blob/master/packages/ember-data/lib/serializers/json-api-serializer.js#L179

**Comment by bmac** [comment]
@john-griffin could your share an example of your payload?

**Comment by john-griffin** [comment]
@bmac here is a sample of the response

``` json
{
  "data": {
    "id": "29039571-e1f3-418c-8889-31b49951ad96",
    "type": "users",
    "links": {
      "self": "https://www.example.com/api/v3/users/29039571-e1f3-418c-8889-31b49951ad96"
    },
    "attributes": {
      "email": "foo@bar.com",
      "first-name": "Foo",
      "last-name": "Bar",
    },
    "relationships": {
      "photo": {
        "links": {
          "self": "https://www.example.com/api/v3/users/29039571-e1f3-418c-8889-31b49951ad96/relationships/photo",
          "related": "https://www.example.com/api/v3/users/29039571-e1f3-418c-8889-31b49951ad96/photo"
        },
        "data": {
          "type": "photos",
          "id": "c3412c5b-9666-413b-b6be-3c4f9988e6a5"
        }
      }
    }
  }
}
```

**Comment by bmac** [comment]
Hi @john-griffin I tried to put together a jsbin using the sample response you provided and I was unable to reproduce the error. Do you mind taking a look at it and let me know what assumption I have wrong?
http://emberjs.jsbin.com/xahori/5/edit?html,js,output

Thanks.

**Comment by john-griffin** [comment]
@bmac thanks for breaking that down in a jsbin. The only difference there was ember-data version. So I upgraded from `1.13.0` to `2.0.0+canary.111a002fe4` and it was fixed. Thanks for your help!

**Comment by wecc** [comment]
For future reference: This was caused by the JSON-API adapter/serializer not opting into the new Serializer API by default. Was fixed by #3375 and in 1.13.3.
